src/anemoi/models/layers/block.py

```diff
@@ -80,6 +81,12 @@ def __init__(
 
         self.layer_norm1 = nn.LayerNorm(num_channels)
 
+        self.register_buffer("grid_lat_coslon_sinlon", grid_lat_coslon_sinlon)
+        if self.grid_lat_coslon_sinlon is not None:
+            self.pos_embedder = nn.Linear(
+                3, num_channels
```

Comment:
instead of `3`, it could be better to use `grid_lat_coslon_sinlon.shape[...]` instead? (not quite sure what the expected shape of that tensor is)

Reply:
Happy to make all this more flexible. I will make a list of points that need to be resolved.

Reply:
Thank you for spotting this. Resolved by [df25841](https://github.com/ecmwf/anemoi-models/pull/96/commits/df258416c87f88a7cf7d90076c241ea26a6a3dc3) and allowed for different ways to calculate the positional encodings..